app/Http/Controllers/Staff/DonationController.php

```diff
@@ -74,16 +74,31 @@ public function update(Request $request, int $id): \Illuminate\Http\RedirectResp
         $donation->status = ModerationStatus::APPROVED;
         $donation->starts_at = $now;
 
-        if ($donation->package->donor_value > 0) {
-            $donation->ends_at = $now->addDays($donation->package->donor_value);
-        } else {
+        if ($donation->package->donor_value === null) {
             $donation->ends_at = null;
+        } else {
+            $activeDonations = Donation::where('user_id', $donation->user_id)
+                ->where('status', ModerationStatus::APPROVED)
+                ->where('ends_at', '>', $now)
+                ->get();
```

Comment:
If a user has 10000+ donations, you don't need them all. You only need the one that has the max ends_at (and to check if it's not null).

Reply:
Oops I reviewed this twice